build.rs

```diff
@@ -20,7 +20,7 @@ fn rerun_if_changed(path_str: &str) -> bool {
 
 fn main() {
     let files_stdout = std::process::Command::new("git")
-        .args(&["ls-tree", "--name-only", "HEAD"])
+        .args(&["ls-tree", "-r", "--name-only", "HEAD"])
```

Comment:
Is this `-r` dispensable?

Reply:
The behavior is undefined when adding a directory to `build-if-changed`. Some system may report the directory has changed when any file has changed, some only when new files are added or existing files are deleted. So it better to list all the files instead.